### ISSUE TITLE
What do I do next?

### ISSUE THREAD

**Issue by stewartbryson** [evryfs/github-actions-runner-operator#122]
I successfully deployed the Helm chart, and then I created a CRD resource and applied it using `kubectl apply -f runner-resource.yaml`. So what's next? Do I need to add a runner in the GitHub UI? Or does the runner automatically startup when the `/workflows/config.yml` references a specific runner name?

Sorry... this is my first time with GitHub self-hosted runners. Thank you.

**Comment by davidkarlsen** [comment]
You should see the runner(s) appearing in https://github.com/organizations/yourorg/settings/actions according to the min/max: https://github.com/evryfs/github-actions-runner-operator/blob/master/config/samples/garo_v1alpha1_githubactionrunner.yaml#L8

**Comment by stewartbryson** [comment]
Thanks for the help @davidkarlsen. I'm using a personal access token value as the secret, and personal access tokens don't seem to have the option to register/deregister runners as one of its permission options. I assume a personal access token is not what I want. Do you have more clarity on this?

**Comment by davidkarlsen** [comment]
A PAT is OK for this - but it needs to have the required scopes in order to be able to register runners. See https://docs.github.com/en/free-pro-team@latest/rest/reference/actions#self-hosted-runners for reference

**Comment by stewartbryson** [comment]
I'm expecting to see a self-hosted runner as soon as the pod is started and available... let me know if that is incorrect.

There are no errors in the log, the pod is available, but I never see a runner in GitHub Actions.

If you have time to look at my config, I'd appreciate it, but understand if you are busy.

https://github.com/RedPillAnalytics/gke-cluster

Thanks for all your help so far.

**Comment by davidkarlsen** [comment]
that looks correct.
What does `kubectl logs runner-pool-<thepodid> -c runner` say?

**Comment by stewartbryson** [comment]
I don't have a pod with a name like `runner-pool-<thepodid>`, only one with a name like `github-actions-runner-operator-<thepodid>`. Am I supposed to have 2 pods? If so... applying my resource is not spinning up the second pod.

**Comment by davidkarlsen** [comment]
yes - the operator will create those pods, let's look in the operator pod then - what does the logs say...?

**Comment by stewartbryson** [comment]
This error:

`2020-12-04T14:47:14.798Z        ERROR   controller      Reconciler error        {"reconcilerGroup": "garo.tietoevry.com", "reconcilerKind": "GithubActionRunner", "contr
oller": "githubactionrunner", "name": "runner-pool", "namespace": "actions-runner", "error": "Secret \"actions-runner\" not found"}`

**Comment by stewartbryson** [comment]
I wasn't adding the secret to the `actions-runner` namespace. That has been corrected. But I now get a new error:
```
2020-12-04T16:06:56.728Z        ERROR   controllers.GithubActionRunner  error from github api   {"githubactionrunner": "actions-runner/runner-pool", "error": "GET https
://api.github.com/orgs/RedPillAnalytics/actions/runners?per_page=30: 401 Bad credentials []"}
```
I'm getting the token from the self-runner API, and I get the token API payload back:
```
{
  "token": "<redacted>",
  "expires_at": "2020-12-04T12:12:37.216-05:00"
}
```

I'm still looking at this, but just wanted to update.

**Comment by stewartbryson** [comment]
I've gotten further. It's the personal access token, not the runner token, that we need to store in the secret. I now have 2 pods. I'll keep you posted. Thanks for the help.

**Comment by stewartbryson** [comment]
I am now hitting the issue in #105 related to `MountVolume.SetUp failed for volume "settings-xml" : configmap "settings-xml" not found`. That issue mentions a solution, but doesn't specify it.

**Comment by stewartbryson** [comment]
@davidkarlsen I meant to say thanks for your help prior to closing.

**Comment by davidkarlsen** [comment]
Well it all depends on your build env requirements - in my case this is a configmap I define for [maven settings xml](https://maven.apache.org/settings.html) - if you don't need it you can just remove it - and do any other changes to the build-pod spec.

**Comment by stewartbryson** [comment]
Yeah... I figured that out. Thanks.
